Flagsmith.FlagsmithClient/AnalyticsProcessor.cs

```diff
@@ -18,7 +19,7 @@ public class AnalyticsProcessor : IAnalyticsProcessor
         readonly string _EnvironmentKey;
         readonly int _TimeOut;
         DateTime _LastFlushed;
-        protected Dictionary<string, int> AnalyticsData;
+        protected ConcurrentDictionary <string, int> AnalyticsData;
```

Comment:
Though this switch to `ConcurrentDictionary` will fix the exception caused by concurrent operations on the plain dictionary, the logic here is still incorrect:
![image](https://github.com/Flagsmith/flagsmith-dotnet-client/assets/1560515/eb27dc5d-ef6e-4a63-a0e2-478906500249)
In the concurrent world you can't rely on simple increments like this `value = value + 1`, it does not give you a proper numbers. You need to rethink how you work with concurrency in this class.
Also you AnalyticsData is protected so other classes in your project might do some mess, it shiuld be private.

Reply:
While the original idea seems to be to avoid real parallel processing by using Flagsmith client as a Singleton and trying to orchestrate other components from there, currently that is not guaranteed for the TrackFeaturemethod of  AnalyticsProcessor class. I think that the best solution is refactoring the Analytics support multithreading, which I did in this PR https://github.com/Flagsmith/flagsmith-dotnet-client/pull/95.